### ISSUE TITLE
Lint/DuplicateMethods false-positive for dynamically-created classes inside methods

### ISSUE THREAD

**Issue by owst** [rubocop-hq/rubocop#5201]
RuboCop seems to be confused by dynamically-defined classes inside methods (i.e. defined using `Class.new`), specifically for methods defined on the meta-class of the dynamically-defined class:
```
# frozen_string_literal: true

def build_class
  Class.new do
    class << self
      def foo; end
    end
  end
end

def build_a_different_class_with_the_same_method_name
  Class.new do
    class << self
      # Flagged as duplicate by RuboCop
      def foo; end
    end
  end
end
```

Seems related, but slightly different to https://github.com/bbatsov/rubocop/issues/5022

--------

## Expected behavior

No duplicate method warning as the methods are defined on distinct classes.

## Actual behavior

```
Inspecting 1 file
W

Offenses:

duplicate_methods_test.rb:15:7: W: Lint/DuplicateMethods: Method .foo is defined at both duplicate_methods_test.rb:6 and duplicate_methods_test.rb:15.
      def foo; end
      ^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

Create `duplicate_methods_test.rb` containing the above two methods and run RuboCop.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.2, running on ruby 2.3.3 x86_64-darwin14)
```

**Comment by Drenmi** [comment]
Closing as duplicate of #5022.
